### ISSUE TITLE
Question...

### ISSUE THREAD

**Issue by MYeager1967** [jonthornton/jquery-timepicker#375]
I am trying to figure out how to display the current time that is assigned to a task in the input box that allows the value to be changed. I an pulling the value into a #variable from a sqlite3 database but can't figure out how to have it show in the selection box. I have the input box itself working though. Is there a way that the timepicker can validate input as valid based on allowed options or is that a task left to the programmer?

If there's a better way to ask questions like this, please let me know.

Thank you for your help,

Mike

**Comment by jonthornton** [comment]
Hey Mike,

It sounds like your problem is related to hooking your database up to the input. I'm afraid I can only help with timepicker-specific stuff.

Timepicker does do validations. It'll trigger a 'timeFormatError' or 'timeRangeError' event if an invalid time is entered. It's up to the programmer to do something with that event.

**Comment by MYeager1967** [comment]
I have the data pulled in from the database, I'm just having issues with
actually getting it into the box. Will send a code example tomorrow.  I'm
sure it's easy enough as I am able to do it with another input selection
box.

**Comment by MYeager1967** [comment]
This line : 

 

<p><td colspan="2">Pool Timer Start Time : <span id="starttime"> </span>  <input id="timeron" type="text" data-time-format="H:i" class="time"/>

<input value="Set" onclick="doit('pool','timeron=' + this.form.timeron.value)" type="button"><br>

 

puts the data from the database (assigned to the starttime variable) right before the input box for timepicker. It works as it’s supposed and the timepicker is fully functional. I know the data is there because it is on the screen. The second line just gives me a button to set the time and pass it to the database. It’s probably possible to do it right from the timepicker but I’m not there yet. What I want to do is have it in the timepicker box instead of in front of it. I have several other selection boxes where I have used the code :

 

$(document).ready(function(){

        $("#poolsp").attr("value",Number(data.poolsetpoint));

    });

 

To assign the data to the value attribute. Then I include a value=”whatever” and it substitutes the data in place, showing it in the box. That doesn’t appear to work here unless I’m somehow messing up the assignment but I’m doing it the same as my other boxes. I’m digging through the sample code (demo) page but not having much luck there. To be honest, I know enough Java to be a danger to myself so I’m learning more as I go. Lots of fun learning new things on an almost regular basis.

 

Here’s the code that actually instantiates the timepicker in case I need to change something here. Is it possible to use the same timepicker code (below) for more than one input box?

 

$(function() {

         $('#timeron').timepicker();

     });

 

Thanks for your help.

 

Mike

**Comment by jonthornton** [comment]
Don't use `$.attr()` to set the value; `$.val()` is safer. I can't tell if that's the cause of your problems, but it's good practice anyway.

Timepicker can initialize many inputs at once. This:
```javascript
$('input.time').timepicker({...options go here...});
```
will initialize timepickers for all inputs with the class `time`.

**Comment by MYeager1967** [comment]
Like so? -> $("#timerondata").val(text(data.timeron));

**Comment by jonthornton** [comment]
Yes.

**Comment by MYeager1967** [comment]
That doesn’t seem to get it either. Here’s a code snippet that covers the jist of what I’m trying to accomplish. Maybe you can get it to work. Not sure what exactly is different but I can’t even get it to display the start time as it does on my page. :( I’ll keep plugging at it... Maybe I’ll figure it out before you prod me in the right direction… I included all the files that make it work in the zip file. It’s a seriously stripped down code snippet. Nothing but the input box and support files.

**Comment by MYeager1967** [comment]
I managed to figure out how to get a static time to pop up in the box by adding this to the code.

 

$(function() {

          $('#timeron').timepicker();

                $('#timeron').timepicker('setTime', '12:00');

});

 

All I need to figure out now is how to put a variable into the ’12:00’ spot.

 

Mike

**Comment by MYeager1967** [comment]
I wanted to give you an update. I got the current value to display where I
wanted. I am using an event to process the changes. Is there a wiki or
something that demonstrates the validation events?

Mike

**Comment by jonthornton** [comment]
No, there's no wiki for that. Catching validation errors is just like listening for change events though.

```javascript
$('#someInput').on('timeFormatError', function(e) {
    // provide some feedback to the user
});
```

**Comment by MYeager1967** [comment]
Will this catch all errors such as someone trying to input text characters?
Also, if I'm catching the change event and there's an error, does the error
fire first so I can avoid processing the change? I'm learning Java as I go
to some extent. Getting better but there's too many poor examples out there
to look over.

Mike

**Comment by jonthornton** [comment]
Yes and yes.

**Comment by MYeager1967** [comment]
I’m using the changeTime event and if I read correctly, it only fires when a valid time is entered. If this is correct, then I already have what I need. Love the picker. It is exactly what I was looking for and I looked over quite a few of them. I realize I could have done it with an <option> setup but this is quite a bit cleaner. I just figured out <option> the other day. Using it for a few other things. I am building an interface for an Arduino Mega powered pool control system so the time picker was nice to be able to set start and stop times. Is there a way to narrow the width of the input box down to something just a little wider that what is needed? I’m using military time and the window is huge.  :)

 

Mike
